rdagent/scenarios/data_science/proposal/exp_gen/base.py

```diff
@@ -0,0 +1,132 @@
+from abc import abstractmethod
+from typing import Literal
+
+from rdagent.core.evolving_framework import KnowledgeBase
+from rdagent.core.proposal import ExperimentFeedback, Hypothesis, Trace
+from rdagent.scenarios.data_science.experiment.experiment import COMPONENT, DSExperiment
+from rdagent.scenarios.data_science.scen import DataScienceScen
+
+
+class DSExpGenCls:
```

Comment:
Why do we create a new class intead of reuse previous class here?

Reply:
because it's a dynamic routing, we don't know which module to initialize each time we call gen function